tss-esapi/src/abstraction/transient/key_attestation.rs

```diff
@@ -0,0 +1,208 @@
+// Copyright 2021 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use super::{ObjectWrapper, TransientKeyContext};
+use crate::{
+    abstraction::ek,
+    constants::SessionType,
+    handles::{AuthHandle, KeyHandle, SessionHandle},
+    interface_types::{
+        algorithm::{AsymmetricAlgorithm, HashingAlgorithm},
+        session_handles::{AuthSession, PolicySession},
+    },
+    structures::{EncryptedSecret, IDObject, SymmetricDefinition},
+    tss2_esys::{Tss2_MU_TPMT_PUBLIC_Marshal, TPM2B_PUBLIC, TPMT_PUBLIC},
+    utils::PublicKey,
+    Error, Result, WrapperErrorKind,
+};
+use log::error;
+use std::convert::{TryFrom, TryInto};
+
+#[derive(Debug)]
+/// Wrapper for the parameters needed by MakeCredential
+pub struct MakeCredParams {
+    /// TPM name of the object
+    pub name: Vec<u8>,
+    /// Encoding of the public parameters of the object whose name
+    /// will be included in the credential computations
+    pub public: Vec<u8>,
```

Comment:
For name it's true, I could just return `Name`, and let the user deserialize, but they're both intended for sending across to a 3rd party to check. And `public` isn't exactly the same as `Public` - it's the marshaled version of the `TPMT_PUBLIC` inside `TPM2B_PUBLIC`, so it'd be a bit of low-level work for users, which doesn't really align well with this abstraction level. `public` isn't _strictly_ required to compute `MakeCredential`, it's more like helper data, meant for the 3rd party to be able to easily verify that the object they're creating the credential for is the correct one.

Reply:
Okay, the reasoning is sound. Please leave name as is but maybe it would be a good idea to enrich the doc comment mentioning what you said, e.g. `Wrapper for the parameters needed by MakeCredential intended for sending across to a 3rd party to check` (or something similar). (A similar API exists for U2F attestations so I guess people will know why it's bytes if they see it's for sending...)

Thanks!

Reply:
Done